virttest/env_process.py

```diff
@@ -560,6 +560,25 @@ def preprocess(test, params, env):
     :param env: The environment (a dict-like object).
     """
     error_context.context("preprocessing")
+
+    # For KVM to work in Power8 systems we need to have SMT=off
+    # and it needs to be done as root, here we do a check whether
+    # we satisfy that condition, if not try to make it off
+    # otherwise throw TestError with respective error message
+    cmd = "grep cpu /proc/cpuinfo | awk '{print $3}' | head -n 1"
```

Comment:
@xutian This suppose to happen before even tests, and should be common for both libvirt and qemu, it would be better if we do it here so that, it will not add any dependencies problem for other APIs.